src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -93,6 +101,47 @@ public async Task<DataTable> ExecuteQuery(string query, string stampName, string
             return await _kustoClient.ExecuteQueryAsync(Helpers.MakeQueryCloudAgnostic(_kustoMap, query), _kustoMap.MapCluster(cluster) ?? cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName);
         }
 
+        public async Task<DataTable> ExecuteQueryOnAllAntaresClusters(string query, string requestId = null, string operationName = null)
+        {
+            if (string.IsNullOrWhiteSpace(operationName))
+            {
+                throw new ArgumentNullException(nameof(operationName), "OperationName cannot be empty. Please supply a name to identify the query.");
+            }
+            if (string.IsNullOrWhiteSpace(query))
+            {
+                throw new ArgumentNullException(nameof(query), "Query cannot be empty. Please supply a query to execute.");
+            }
+            else
+            {
+                List<Task<DataTable>> queryTask = new List<Task<DataTable>>();
+
+                if(isPublicCloud)
+                {
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-bay-153", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|WestUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-blu-189", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|EastUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-dm1-187", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|CentralUS"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-am2-329", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|WestEurope"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-db3-169", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|NorthEurope"));
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-hk1-029", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|EastAsia"));
+                }
+                else
+                {
+                    queryTask.Add(ExecuteQuery(query, DataProviderConstants.FakeStampForAnalyticsCluster, null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|{_configuration.CloudDomain}"));
+                }
+
+                var queryResult = await Task.WhenAll(queryTask.ToArray());
+                DataTable mergedTable = new DataTable();
+                foreach (DataTable dt in queryResult)
+                {
+                    if (dt.Rows.Count > 0)
+                    {
+                        mergedTable.Merge(dt, false, MissingSchemaAction.Add);
```

Comment:
>mergedTable.Merge [](start = 24, length = 17)

We may end up spiking RuntimeHost CPU if someone writes a really bad query which has a lot of records. Do you want to put some hard limits here? Like 10K records or 100K records?

Reply:
Does 3K rows per cluster sound reasonable or do you think the limit should be even lower. I am thinking 3K since internally we allow people to run query for a 3 day duration.

Reply:
Yea sounds fine. We can bump it or lower it once we know how this behaves

Reply:
@puneetg1983 , I looked up the statistics and I do not see any query in the past 2 days crossing 2K rows when using All, I believe 3K limit should be good to avoid queries from failing however like you said, we need to keep an eye on the CPU and adjust the limit accordingly.